### PR TITLE
Toolbars prepare part4

### PR DESCRIPTION
**Pull request by JulianWielga**

### CODE REVIEW

File: ui/client/components/ToolBox.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  const processDefinitionData = useSelector(getProcessDefinitionData)
+  const processCategory = useSelector(getProcessCategory)
+
+  const nodesToAdd = useMemo(
+    () => ProcessDefitionUtils.getNodesToAddInCategory(processDefinitionData, processCategory),
+    [processDefinitionData, processCategory],
+  )
+
+  const nodeGroupIsEmpty = (nodeGroup) => {
+    return nodeGroup.possibleNodes.length == 0
```
Comment (PPiedel) [inline ui/client/components/ToolBox.tsx]: could be inlined

File: ui/client/store/configureStore.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/* eslint-disable i18next/no-literal-string */
```
Comment (arkadius) [inline ui/client/store/configureStore.ts]: In this change you only rewritten this class to TS or it was changed anyway?

File: ui/client/components/ToolBox.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    if (!nodeGroupIsEmpty(nodeGroup)) {
+      dispatch(toggleToolboxGroup(nodeGroup.name))
+    }
+  }
+
+  return (
+    <div id="toolbox">
+      <div>
+        {nodesToAdd.map(nodeGroup => {
+          const label =
```
Comment (PPiedel) [inline ui/client/components/ToolBox.tsx]: Maybe it should be separated component, with "label" logic inside it?

File: ui/client/components/graph/GraphContext.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable i18next/no-literal-string */
+import React, {createContext, PropsWithChildren, useContext} from "react"
+import Graph from "./Graph"
+
+const GraphContext = createContext<() => Graph>(null)
```
Comment (PPiedel) [inline ui/client/components/graph/GraphContext.tsx]: Maybe we could extract () => Graph as separate type?

File: ui/client/components/graph/GraphContext.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable i18next/no-literal-string */
```
Comment (alien11689) [inline ui/client/components/graph/GraphContext.tsx]: why is the whole file check disabled when it is needed only by one line?

File: ui/client/store/configureStore.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  const store = createStore(
+    persistReducer(persistConfig, reducer),
+    composeWithDevTools(
+      applyMiddleware(analyticsMiddleware, thunk),
+    ),
+  )
+  const persistor = persistStore(store)
+
+  if (module.hot) {
+    // eslint-disable-next-line i18next/no-literal-string
```
Comment (alien11689) [inline ui/client/store/configureStore.ts]: can you fix it by extracting '../reducers' to constant `moduleName`?

File: ui/client/components/ToolBox.tsx [hunk 7]
```diff
@@ -0,0 +1,66 @@
+import React, {useMemo} from "react"
```
[the remark this answers is not in the document]
Comment (JulianWielga) [reply ui/client/components/ToolBox.tsx]: no changes except simpler redux connection with hooks

File: ui/client/store/configureStore.ts (hunk 2 above)
Comment (JulianWielga) [reply ui/client/store/configureStore.ts]: like in commit message, added state persistency

File: ui/client/components/ToolBox.tsx (hunk 1 above)
Comment (JulianWielga) [reply ui/client/components/ToolBox.tsx]: it's not in scope of my change.

File: ui/client/components/ToolBox.tsx (hunk 3 above)
Comment (JulianWielga) [reply ui/client/components/ToolBox.tsx]: it's not in scope of my change.

File: ui/client/components/graph/GraphContext.tsx (hunk 4 above)
Comment (JulianWielga) [reply ui/client/components/graph/GraphContext.tsx]: what will we get from it?

**Pull request merged**